Framework/Kernel/inc/MantidKernel/FacilityInfo.h

```diff
@@ -95,6 +95,9 @@ class MANTID_KERNEL_DLL FacilityInfo {
   /// Returns the catalogInfo class.
   const CatalogInfo &catalogInfo() const { return m_catalogs; }
 
+  /// Returns a bool indicating whether prefix is required in file names
+  const bool &allowNoPrefix() const { return m_allowNoPrefix; }
```

Comment:
Does returning a bool by const reference make sense? Further down, it looks like you are using the result immediately.